crates/sui-bridge/src/client/bridge_client.rs

```diff
@@ -3,34 +3,42 @@
 
 //! `BridgeClient` talks to BridgeNode.
 
+use std::str::FromStr;
 use std::sync::Arc;
 
 use crate::crypto::{verify_signed_bridge_action, BridgeAuthorityPublicKeyBytes};
 use crate::error::{BridgeError, BridgeResult};
 use crate::server::APPLICATION_JSON;
 use crate::types::{BridgeAction, BridgeCommittee, VerifiedSignedBridgeAction};
+use url::Url;
 
+// Note: `base_url` is `Option<Url>` because `quorum_map_then_reduce_with_timeout_and_prefs`
+// uses `[]` to get Client based on key. Therefore even when the URL is invalid we need to
+// create a Client instance. In the future we can consider change
+// `quorum_map_then_reduce_with_timeout_and_prefs` and its callsites to use `get` instead of `[]`.
 #[derive(Clone, Debug)]
 pub struct BridgeClient {
     inner: reqwest::Client,
     authority: BridgeAuthorityPublicKeyBytes,
     committee: Arc<BridgeCommittee>,
-    base_url: String,
+    base_url: Option<Url>,
 }
 
 impl BridgeClient {
-    pub fn new<S: Into<String>>(
-        base_url: S,
-        authority: BridgeAuthorityPublicKeyBytes,
+    pub fn new(
+        authority_name: BridgeAuthorityPublicKeyBytes,
         committee: Arc<BridgeCommittee>,
     ) -> BridgeResult<Self> {
-        if !committee.is_active_member(&authority) {
-            return Err(BridgeError::InvalidBridgeAuthority(authority));
+        if !committee.is_active_member(&authority_name) {
+            return Err(BridgeError::InvalidBridgeAuthority(authority_name));
         }
+        let member = committee
+            .member(&authority_name)
+            .expect("authority is active member");
```

Comment:
not sure i understand, how is `unwrap` different from `expect` here? or you are simply suggesting to add a comment that we checked is_active_member above?

Reply:
Very minor, but my suggestion was because there is no chance of the `expect()` being hit and needing the custom message because of the check in `is_active_member` so an `unwrap()` makes more sense because it will work 100% of the time